kernel_gateway/gatewayapp.py

```diff
@@ -528,5 +530,8 @@ def shutdown(self):
         """Stop all kernels in the pool."""
         self.personality.shutdown()
 
+    def _signal_stop(self, sig, frame):
+        self.log.info("Received signal to terminate.")
+        ioloop.IOLoop.current().stop()
```

Comment:
Excellent feedback - thank you for catching this.  I've tested the recommended changes and posted the commit.